db_benchmark_test.go

```diff
@@ -109,30 +187,156 @@ func setup(b *testing.B) *sql.DB {
+	// 10| event_type_id     | integer                     |           |
+
+	orgID := i % 1000                  // limited number of org IDs
+	accountNumber := orgID + 1         // can be different than org ID
+	clusterName := uuid.New().String() // unique
+	notificationTypeID := 1            // instant report
+	stateID := 1 + i%4                 // just four states can be used
+	updatedAt := time.Now()            // don't have to be unique
+	notifiedAt := time.Now()           // don't have to be unique
+	errorLog := ""                     // usually empty
+	eventTypeID := i%2 + 1             // just two event type targets possible
+
+	_, err := connection.Exec(insertIntoReportedV2Statement, orgID,
+		accountNumber, clusterName, notificationTypeID, stateID,
+		report, updatedAt, notifiedAt, errorLog, eventTypeID)
+
+	// check for any error, possible to exit immediatelly
+	if err != nil {
+		b.Fatal(err)
+	}
+}
+
+// runBenchmarkInsertIntoReportedTableV1 function perform several inserts into
+// reported table v1. In case of any error detected, benchmarks fail
+// immediatelly.
+func runBenchmarkInsertIntoReportedTableV1(b *testing.B, scaleFactor int, report *string) {
+	// retrieve DB connection
 	connection := setup(b)
 	if connection == nil {
 		b.Fatal()
 	}
 
+	// make sure we start with no DB table
+	execSQLStatement(b, connection, dropTableReportedV1)
+
+	// create reported table from scratch
+	execSQLStatement(b, connection, createTableReportedV1)
+
+	// good citizens cleanup properly
+	//defer execSQLStatement(b, connection, dropTableReportedV1)
+
+	// time to start benchmark
 	b.ResetTimer()
 
 	// perform DB benchmark
 	for i := 0; i < b.N; i++ {
+		for j := 0; j < scaleFactor; j++ {
+			insertIntoReportedV1(b, connection, i, report)
+		}
 	}
 }
 
-// BenchmarkBar is dummy ATM
-func BenchmarkBar(b *testing.B) {
+// runBenchmarkInsertIntoReportedTableV2 function perform several inserts into
+// reported table v2. In case of any error detected, benchmarks fail
+// immediatelly.
+func runBenchmarkInsertIntoReportedTableV2(b *testing.B, scaleFactor int, report *string) {
 	connection := setup(b)
 	if connection == nil {
 		b.Fatal()
 	}
 
+	// make sure we start with no DB table
+	execSQLStatement(b, connection, dropTableReportedV2)
+
+	// create reported table from scratch
+	execSQLStatement(b, connection, createTableReportedV2)
+
+	// good citizens cleanup properly
+	//defer execSQLStatement(b, connection, dropTableReportedV1)
+
+	// time to start benchmark
 	b.ResetTimer()
 
 	// perform DB benchmark
 	for i := 0; i < b.N; i++ {
+		for j := 0; j < scaleFactor; j++ {
+			insertIntoReportedV2(b, connection, i, report)
+		}
 	}
 }
+
+// BenchmarkInsertIntoReportedTableV1 checks the speed of inserting into
+// reported table without event_type column
+func BenchmarkInsertIntoReportedTableV1(b *testing.B) {
+	report := ""
+
+	runBenchmarkInsertIntoReportedTableV1(b, 1, &report)
```

Comment:
why is `scaleFactor` 1?

Reply:
We need to run benchmark with tables of different sizes. So changing N via CLI flags won't be sufficient there

Reply:
yes, but why 1 and not 10 f.e?

Reply:
ah sure, as I said, there will be much more benchmarks prepared later (today?)

Reply:
nice!